src/test/java/com/expedia/pitchfork/systems/common/SpanValidatorTest.java

```diff
@@ -35,14 +38,15 @@ public void setup() {
 
     @ParameterizedTest
     @MethodSource("timestamps")
-    public void shouldDiscardSpanIfTimestampIsInvalid(boolean rejectNullTimestamps, int maxDrift, Long timestamp, boolean spanIsKept) {
-        zipkin2.Span zipkinSpan = zipkin2.Span.newBuilder()
+    public void shouldDiscardSpanIfTimestampIsInvalid(boolean rejectNullTimestamps, int maxDrift, Long timestamp, boolean spanIsKept, String[] invalidServiceNames) {
+        zipkin2.Span zipkinSpan = Span.newBuilder()
                 .traceId(zipkinTraceId(123L))
                 .id(zipkinSpanId(456L))
+                .localEndpoint(Endpoint.newBuilder().serviceName("unknown").build())
```

Comment:
This test is named "should discard span if **timestamp** is invalid". But you are providing also an invalid **service name**. Does this mean that the test is not only testing the invalid timestamp?

Reply:
I've split this into 2 separate tests. One for the timestamps and one for the service names.